scripts/compress_images_test.py

```diff
@@ -0,0 +1,261 @@
+# Copyright 2019 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for compress_images.py."""
+
+from __future__ import annotations
+
+import os
+import pathlib
+import shutil
+import tempfile
+import unittest
+from unittest import mock
+
+from scripts import compress_images
+
+from PIL import Image
+from typing import List, TypedDict
+
+
+class CompressedImageInfo(TypedDict):
+    """Type definition for compressed image information."""
+
+    path: pathlib.Path
+    original_size: int
+    new_size: int
+
+
+class TestImageCompressor(unittest.TestCase):
+    """Test the ImageCompressor class."""
+
+    def setUp(self) -> None:
+        """Set up test environment before each test."""
+        self.temp_dir = tempfile.mkdtemp()
+        self.output_dir = os.path.join(self.temp_dir, 'compressed_images')
+
+        self.png_path = os.path.join(self.temp_dir, 'test_image.png')
+        self.jpg_path = os.path.join(self.temp_dir, 'test_image.jpg')
+        self.webp_path = os.path.join(self.temp_dir, 'test_image.webp')
+
+        self._create_test_image(self.png_path, 'PNG')
+        self._create_test_image(self.jpg_path, 'JPEG')
+        self._create_test_image(self.webp_path, 'WEBP')
+
+    def tearDown(self) -> None:
+        """Clean up test environment after each test."""
+        shutil.rmtree(self.temp_dir)
+
+    def _create_test_image(self, path: str, Img_format: str) -> None:
+        """Create a test image for compression testing."""
+        img = Image.new('RGB', (100, 100), color='red')
+        img.save(path, format=Img_format)
+
+    def test_get_compression_type_zip_extension(self) -> None:
+        """Test get_compression_type returns 'Zip' for PNG."""
+        compressor = compress_images.ImageCompressor(self.temp_dir)
+        compression_type = compressor.get_compression_type('.png')
+        self.assertEqual(compression_type, 'Zip')
+
+    def test_get_compression_type_lzw_extension(self) -> None:
+        """Test get_compression_type returns 'LZW' for JPG."""
+        compressor = compress_images.ImageCompressor(self.temp_dir)
+        compression_type = compressor.get_compression_type('.jpg')
+        self.assertEqual(compression_type, 'LZW')
+
+    @mock.patch('subprocess.run')
```

Comment:
please use our `swap` utilities for mocking instead of `mock.patch`. See https://github.com/oppia/oppia/wiki/Backend-tests#backend-test-structure for instructions (in fact, you should really read through that whole wiki page to understand our backend testing conventions). ditto elsewhere